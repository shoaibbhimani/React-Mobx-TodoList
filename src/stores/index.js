import todostore from './todoStore'

export default {
    todostore
}