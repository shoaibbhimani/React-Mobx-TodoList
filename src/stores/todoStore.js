import { extendObservable, action } from 'mobx';
import remotedev from 'mobx-remotedev/lib';

class Todos {
  constructor(){
  	extendObservable(this,{
	    todos: [{ name:"foo", completed:false}, { name:"bar", completed:true}],
	    activeEditContent:null,
	    editTodo:action((index) => {
	      let todos = this.todos.slice();
	      todos[index].index = index;
	      this.activeEditContent = Object.assign({},todos[index]);
	    }),
	    makeChange:action((newContentName) => {
    	 //remove Previous content from todos
    	 let todos = this.todos.slice();
    	
    	 //make new todo
    	 let newContentObject = {};
    	 newContentObject['name'] = newContentName;
    	 newContentObject['completed'] = false;

    	 todos[this.activeEditContent.index] = newContentObject;

    	 this.todos = todos;
    	 this.activeEditContent = null;
	    }),
	    addTodo:action((name) => {
	     if(!name){
	     	alert("Please enter Something");
	     	return;
	      }
	      let newTodo = {};
	      newTodo.name = name;
		    newTodo.completed = false;
	      this.todos = this.todos.concat(newTodo)
	    }),
	    get getTodos() {
          return this.todos
        },
        get completedTodos(){
         return this.todos.filter((todo) => todo.completed)
        },
	    removeTodo:action((index) => {
	      let todos = this.todos.slice();
	      todos.splice(index,1);
	      this.todos = todos;
	    }),
	    completed:action((index) => {
	      let todos = this.todos.slice();
	      todos[index].completed =  !todos[index].completed
	      this.todos = todos;
	    }),
	    changeContent:action((index, content) => {
	      let todos = this.todos.slice();
	      todos[index].name =  content
	      this.todos = todos;
	    })
   })
  }   
}


export default remotedev(new Todos());

export {
 Todos
}