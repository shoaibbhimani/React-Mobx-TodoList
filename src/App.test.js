import React from 'react';
import { Todos } from './stores/todoStore'

describe("TodoStore", function(){
 beforeEach(function() {
   this.store = new Todos();
 });
  // Add Data to list
  it('Add to in a List', function(){
   this.store.addTodo({ name:'Todo1', completed:false });
   expect(this.store.todos.length).toBe(3);
  });

  it('Remove to in a List', function(){
     this.store.removeTodo(0);
     expect(this.store.todos[0].name).toBe("bar");
  });
});

