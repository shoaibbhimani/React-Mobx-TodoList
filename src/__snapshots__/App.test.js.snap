exports[`TodoStore Add to in a List 1`] = `3`;

exports[`TodoStore Remove to in a List 1`] = `"bar"`;
